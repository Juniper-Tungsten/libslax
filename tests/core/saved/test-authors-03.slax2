/* Machine Crafted with Care (tm) by slaxWriter */
version 1.1;

match / {
    <out> {
        /* Show the oldest five deceased authors */
        
        for-each (/authors/author) {
            sort life-span/died - life-span/born {
                order "descending";
            }
            sort life-span/born;
            if (position() <= 5) {
                var $age = life-span/died - life-span/born;
                
                <oldest age=$age> {
                    copy-of .;
                }
            }
        }
    }
}
