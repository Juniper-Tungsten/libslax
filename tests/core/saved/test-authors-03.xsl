<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
  <xsl:template match="/">
    <out>
      <!-- Show the oldest five deceased authors -->
      <xsl:for-each select="/authors/author">
        <xsl:sort select="life-span/died - life-span/born" order="descending"/>
        <xsl:sort select="life-span/born"/>
        <xsl:if test="position() &lt;= 5">
          <xsl:variable name="age" select="life-span/died - life-span/born"/>
          <oldest age="{$age}">
            <xsl:copy-of select="."/>
          </oldest>
        </xsl:if>
      </xsl:for-each>
    </out>
  </xsl:template>
</xsl:stylesheet>
