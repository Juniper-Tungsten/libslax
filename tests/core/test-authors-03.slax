version 1.0;

match / {
    <out> {
	/* Show the oldest five deceased authors */
    	for-each (/authors/author) {
	    sort life-span/died - life-span/born {
		order "descending";
	    }
	    sort life-span/born;

	    if (position() <= 5) {
                var $age = life-span/died - life-span/born;
		<oldest age = $age> {
                    copy-of .;
		}
	    }
	}
    }
}
