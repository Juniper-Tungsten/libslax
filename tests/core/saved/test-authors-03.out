<?xml version="1.0"?>
<out>
  <oldest age="NaN">
    <author>
      <name>
        <first>Emery</first>
        <last>Bonett</last>
      </name>
    </author>
  </oldest>
  <oldest age="NaN">
    <author>
      <name>
        <first>John</first>
        <last>Bonett</last>
      </name>
    </author>
  </oldest>
  <oldest age="NaN">
    <author>
      <name>
        <first>Lilian Jackson</first>
        <last>Braun</last>
      </name>
    </author>
  </oldest>
  <oldest age="NaN">
    <author>
      <name>
        <first>Robert</first>
        <last>Crais</last>
      </name>
    </author>
  </oldest>
  <oldest age="NaN">
    <author>
      <name>
        <first>Jordan</first>
        <last>Dane</last>
      </name>
    </author>
  </oldest>
</out>
